src/dbus/dbus_object.cpp

```diff
@@ -52,18 +54,41 @@ otbrError DBusObject::Init(void)
 
     VerifyOrExit(dbus_connection_register_object_path(mConnection, mObjectPath.c_str(), &vTable, this),
                  err = OTBR_ERROR_DBUS);
+    RegisterMethod(kDBusPropertyInterface, kDBusPropertyGetMethod,
+                   std::bind(&DBusObject::GetPropertyMethodHandler, this, _1));
+    RegisterMethod(kDBusPropertyInterface, kDBusPropertySetMethod,
+                   std::bind(&DBusObject::SetPropertyMethodHandler, this, _1));
+    RegisterMethod(kDBusPropertyInterface, kDBusPropertyGetAllMethod,
+                   std::bind(&DBusObject::GetAllPropertiesMethodHandler, this, _1));
 exit:
     return err;
 }
 
-void DBusObject::RegisterMethod(const std::string &aInterfaceName,
-                                const std::string &aMethodName,
-                                const HandlerType &aHandler)
+void DBusObject::RegisterMethod(const std::string &      aInterfaceName,
+                                const std::string &      aMethodName,
+                                const MethodHandlerType &aHandler)
 {
     std::string fullPath = aInterfaceName + "." + aMethodName;
 
-    assert(mHandlers.find(fullPath) == mHandlers.end());
-    mHandlers.emplace(fullPath, aHandler);
+    assert(mMethodHandlers.find(fullPath) == mMethodHandlers.end());
+    mMethodHandlers.emplace(fullPath, aHandler);
+}
+
+void DBusObject::RegisterGetPropertyHandler(const std::string &        aInterfaceName,
```

Comment:
```suggestion
void DBusObject::RegisterProperty(const std::string &aInterfaceName, const std::string &aPropertyName, const PropertyHandlerType &aGetHandler, PropertyHandlerType &aSetHandler);
```
And define a special NullHandler which means not supported? Thus, the object knows how to generate introspect.